[a11y] Docs for checkbox, radio button, and choice list (#1145)
Shopify/polaris-react: src/components/Checkbox/README.md

See Material Design and development documentation about accessibility for Android:

- [Accessible design on Android](https://material.io/design/usability/accessibility.html)
- [Accessible development on Android](https://developer.android.com/guide/topics/ui/accessibility/)

<!-- /content-for -->

<!-- content-for: ios -->

See Apple’s Human Interface Guidelines and API documentation about accessibility for iOS:

- [Accessible design on iOS](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/accessibility/)
- [Accessible development on iOS](https://developer.apple.com/accessibility/ios/)

<!-- /content-for -->

<!-- content-for: web -->

Screen readers convey the state of the checkbox automatically.

- Use the `disabled` prop to apply the HTML `disabled` attribute to the checkbox `<input>`. This prevents merchants from being able to interact with the checkbox, and conveys its inactive state to assistive technologies.
- Use the `id` prop to provide a unique `id` attribute value for the checkbox. If an `id` isn’t provided, then the component generates one. All checkboxes must have unique `id` values to work correctly with assistive technologies.

> - If you set the `checked` prop to `indeterminate`, then the state of the checkbox is conveyed using `aria-checked=”mixed”`.

Reviewer: ```suggestion
- If you set the `checked` prop to `indeterminate`, then the state of the checkbox is conveyed using `aria-checked="mixed"`.
```

Reviewer: I'm also wondering if this could be simplified into:

```md
- Setting `checked="indeterminate"` conveys the state of the checkbox using `aria-checked="mixed"`.
```

Thoughts?

Author: Works for me!

> - If you set the `checked` prop to `indeterminate`, then the state of the checkbox is conveyed using `aria-checked="mixed"`.

Author: ```suggestion
- Setting `checked="indeterminate"` conveys the state of the checkbox using `aria-checked="mixed"`.
```
